pkg/catalogv2/system/system.go

```diff
@@ -211,13 +212,14 @@ func (m *Manager) Uninstall(namespace, name string) error {
 	return m.waitPodDone(op)
 }
 
-func (m *Manager) Ensure(namespace, name, minVersion string, values map[string]interface{}, forceAdopt bool, installImageOverride string) error {
+func (m *Manager) Ensure(namespace, name, minVersion, exactVersion string, values map[string]interface{}, forceAdopt bool, installImageOverride string) error {
```

Comment:
This will need unit tests.

Reply:
Agreed, but it's hard to test. The design of the struct will need to be changed significantly. The problem with `Ensure` is that it runs asynchronously - returns immediately. We could then test the private methods (which is already not a great solution). Specifically - `install` and `isInstalled`. They would also need to be refactored. I could create independent, small functions that only work with version strings and maybe `values` maps and report whether a version is outdated or not.